entity/c/c_api.c

```diff
@@ -0,0 +1,197 @@
+#include "c_api.h"
+
+extern int sent_seq_num;
+extern unsigned char entity_client_state;
+extern unsigned char entity_server_state;
+extern long int st_time;
+
+config_t *load_config(char *path)
+{
+    return load_config_t(path);
+}
+
+session_key_t *get_session_key(config_t *config_info)
+{
+    if (strncmp((const char *)config_info->network_protocol, "TCP", 3) == 0)
+    {
+        return send_session_key_req_via_TCP(config_info);
+    }
+    else if (strncmp((const char *)config_info->network_protocol, "UDP", 3) == 0)
+    {
+        return send_session_key_req_via_UDP();
+    }
+    return 0;
+}
+
+int secure_connect_to_server(session_key_t *s_key, config_t *config_info)
+{
+    int sock;
+    connect_as_client((const char *)config_info->entity_server_ip_addr, (const char *)config_info->entity_server_port_num, &sock);
+    unsigned char entity_nonce[HS_NONCE_SIZE];
+    unsigned int parsed_buf_length;
+    unsigned char *parsed_buf = parse_handshake_1(s_key, entity_nonce, &parsed_buf_length);
+    unsigned char sender[128]; // TODO: actually only needs 19 bytes.
```

Comment:
What would be the reasonable buffer size here? I recommend using `#define` for all constants if possible.

Reply:
Comment below

Reply:
Your comment below does not seem to answer my question. You (or one of you who contributed to this PR) used 128 and mentioned it actually needs 19 bytes. This really confuses the reader.

Why do you have 128 while needing only 19 bytes? What would be the maximum length of the sender name? If 128 is enough, define a constant and add an explanation there.